[Diwen] iP

StevenLiudw opened the pull request:
# Duke Task Manager

> "For every minute spent organizing, an hour is earned." - Benjamin Franklin

**Duke Task Manager** helps you keep track of all your tasks. With **Duke**, you can:
- Store tasks
- Retrieve tasks
- Complete tasks

To get started with **Duke**, simply follow these steps:

1. Clone the repository.
2. Compile the source files.
3. Run the main method.
4. Begin adding your tasks!

```java
public class Duke {
    public static void main(String[] args) {
        new Duke().run();
    }
}
```

Features:
- [x] Task adding
- [x] Task deletion
- [x] Mark tasks as done
- [x] Set task deadlines



Need more info on Duke? Check out our [User Guide](https://github.com/duke-repo/user-guide)(Comming soon).

Tip: Duke is case-sensitive. So be sure to input commands exactly as they are given.

Enjoy organizing with Duke!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level 0. Rename, Greet, Exit

Commit:
Level 1. Echo

Commit:
Level 2. Add, List

Commit:
Level 3. Mark as Done ( Create Task class )

Commit:
Level 4. ToDos, Events, Deadlines( Use Inheritance to support multiple task types )

Commit:
Duke A-TextUiTesting: Automated Text UI Testing

Commit:
Revert "Duke A-TextUiTesting: Automated Text UI Testing"

This reverts commit 6245d1ed34434025159caf995f859eadd8497537.

Commit:
Level 5. Handle Errors

Commit:
Level-6: Delete （Add delete function and also add more exception on mark and unmark to avoid the situation that the input number is not valid)

Commit:
A-Enums: Use Enums
1. Use Enums to avoid type error of task type
2. Fix a few bugs in Duke.java
3. Use the ui-test to detect bug

Commit:
Storage.java : Add Storage.java

Commit:
Deadline.java : change the d_time from string to LocalDate.
Event: change the start and end from string to LocalDate.
Make the chatbot understand the format yyyy-MM-dd

Commit:
A-MoreOOP
Ui: deals with interactions with the user
Storage: deals with loading tasks from the file and saving tasks in the file
Parser: deals with making sense of the user command
TaskList: contains the task list e.g., it has operations to add/delete tasks in the list
Command classes
Add JavaDoc for each file

Commit:
A-Packages
Divide classes into packages
Add duke.command
Add duke.storage
Add duke.task
Add duke.ui

Commit:
build.gradle: add checkstyle plugins

Commit:
Merge branch 'add-gradle-support'

Commit:
ParserTest.java: Add a unit test for Paser class
TaskListTest.java: Add a unit test for TaskList class

Commit:
ParserTest.java: add javadoc
TaskListTest.java: add javadoc

Commit:
All classes: Follow the Coding Standard

Commit:
FindCommand.java: Add this file for the new feature"find"

Commit:
checkstyle.xml: add the new checkstyle file
suppressions.xml: add the new file

Commit:
Level-10 Add GUI

Commit:
DukeException.java: Use Java varargs feature

teoks0199 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,59 @@
+ */
+public class Parser {
+
+    /**
```
I like how clear your documentation is!

teoks0199 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+     * @param keyword The search keyword.
+     * @param ui The UI object to display results.
+     */
+    public String findTasks(String keyword, Ui ui) {
```
I like how you name the methods so consistently!

teoks0199 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,43 @@
+     *
+     * @return true if Duke should exit, false otherwise.
+     */
+    public boolean isExit() {
```
Perhaps a better name here could be isFinshed?

teoks0199 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,118 @@
+    }
+
+    /**
+     * Returns a line to format the UI.
```
Should this be "to" instead?

teoks0199 reviewed (commented):
Good job 😄

Commit:
Localization: Add local package for event and deadline classes
User's system language settings can affect the displayed language
in the program. This inconsistency can lead to potential confusion
for users who might be expecting content in English.

To ensure a consistent user experience, it's vital to standardize
the display language.

Let's introduce a local package for the event and deadline classes
to ensure that the program output is always in English, irrespective
of the user's system language settings.

Opting for a local package approach allows us to have better control
over the localization settings without relying on external variables
like system settings.

Commit:
Do a increment for branch-0Level-8

Commit:
Do a increment for the branch-Level-7

StevenLiudw committed:
Merge pull request #3 from StevenLiudw/master

merge pr

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Commit:
Assert feature: Document assumptions in key classes
In complex systems like ours, it's essential to have clear
documentation of the assumptions at different parts of the code.
This not only improves readability but also aids in debugging
potential issues in the future.

To ensure these assumptions are evident and actively checked during
runtime, the assert feature is a suitable tool.

Let's integrate the assert feature into DialogBox.java, Duke.java,
Parser.java, TaskList.java, Event.java, MainWindow.java, and Ui.java.
This will help in conveying and verifying our assumptions in these
critical parts of the code.

Note: I am utilizing the built-in assert feature of Java, not JUnit
assertions, for this purpose.

Commit:
Revert "Assert feature: Document assumptions in key classes"

This reverts commit f3b4d030706f8fb1349e2a02eae7a548320c4150.

Commit:
Assert feature: Document assumptions in key classes
In complex systems like ours, it's essential to have clear
documentation of the assumptions at different parts of the code.
This not only improves readability but also aids in debugging
potential issues in the future.

To ensure these assumptions are evident and actively checked during
runtime, the assert feature is a suitable tool.

Let's integrate the assert feature into DialogBox.java, Duke.java,
Parser.java, TaskList.java, Event.java, MainWindow.java, and Ui.java.
This will help in conveying and verifying our assumptions in these
critical parts of the code.

Note: I am utilizing the built-in assert feature of Java, not JUnit
assertions, for this purpose.

StevenLiudw committed:
Merge pull request #5 from StevenLiudw/heads/A-Assertions

Heads/a assertions

Commit:
Code quality: Refactor for improved maintainability
Maintaining high code quality is paramount for the longevity and
scalability of our project. Over time, as code evolves, it's
crucial to revisit and refactor parts that may have become less
optimal or diverged from best practices.

In this commit, a critical examination of the codebase was
undertaken. Several areas were identified and refactored to enhance
readability, maintainability, and adhere to our coding standards.

These changes not only make the codebase cleaner but also ensure
that future developers can understand and modify the code with
greater ease.

It's important to note that while making these quality improvements,
care was taken to preserve the existing functionality and ensure
backwards compatibility.

StevenLiudw committed:
Merge pull request #6 from StevenLiudw/heads/A-CodeQuality

Code quality: Refactor for improved maintainability

Commit:
Event.java: Integrate Java Streams for data processing
Java Streams provide a modern, declarative approach to data
manipulation, allowing for more readable and concise code.

In `Event.java`, traditional iterative approaches were replaced
with the Streams API. This transition streamlines various
operations, making them more intuitive and reducing the risk
of common iterative errors.

By adopting Java Streams, we're also aligning with current Java
best practices, ensuring that our code remains up-to-date with
contemporary Java development patterns.

StevenLiudw committed:
Merge pull request #7 from StevenLiudw/A-Streams

Event.java: Integrate Java Streams for data processing

StevenLiudw committed:
A-CI

create main.yml for CI

StevenLiudw committed:
Merge pull request #8 from StevenLiudw/A-CI

A-CI

Commit:
C-Tagging Provide a way to tag items

StevenLiudw committed:
Merge pull request #9 from StevenLiudw/branch-Level-9

Merge pull request branch-Level-9

Commit:
A-BetterGui
1. make the color of GUI to be yellow
2. make it resizable

Commit:
A-Personality Change the font for the input to be more robotic and also change the logo for Jarvis

Commit:
A-MoreErrorHandling

Commit:
A-MoreTesting

Commit:
Merge branch 'master' of https://github.com/StevenLiudw/ip

Commit:
Add ui png

StevenLiudw committed:
Update README.md

StevenLiudw committed:
Merge pull request #10 from StevenLiudw/A-UserGuide

Update README.md

StevenLiudw committed:
Update README.md

StevenLiudw committed:
Update README.md

StevenLiudw committed:
Update README.md

StevenLiudw committed:
Update README.md

StevenLiudw committed:
Update README.md

Commit:
Fix some style error

Commit:
Fixed some style errors

Commit:
check the style error again and also fix a bug about addtag
